WordPress/WordPressTest/WordPressAPI/WordPressComOAuthClientTests.swift

```diff
@@ -0,0 +1,125 @@
+import XCTest
+import WordPress
+import OHHTTPStubs
+
+class WordPressOAuthClientTests: XCTestCase {
+
+    let WordPressComOAuthTokenUrl = "https://public-api.wordpress.com/oauth2/token"
+
+    override func setUp() {
+        super.setUp()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        OHHTTPStubs.removeAllStubs()
+    }
+
+    func testAuthenticateUsernameNo2FASuccessCase() {
+        OHHTTPStubs.stubRequestsPassingTest({(request) -> Bool in
+            return request.URL?.absoluteString == self.WordPressComOAuthTokenUrl
+            }, withStubResponse:{ (request:NSURLRequest) -> OHHTTPStubsResponse in
+                let mockDataURL = NSBundle(forClass: self.dynamicType).URLForResource("WordPressOAuthClientSuccess", withExtension:"json")!
+                let mockData = NSData(contentsOfURL: mockDataURL)!
+                let stubResponse = OHHTTPStubsResponse(data: mockData, statusCode:200, headers:["Content-Type":"application/json"])
+                stubResponse.responseTime = OHHTTPStubsDownloadSpeedWifi;
+                return stubResponse
+        })
```

Comment:
OHHTTPStubs has a more concise syntax for Swift now. First, I'd define a private helper to match the request:

```swift
private func isOauthTokenRequest() -> OHHTTPStubsTestBlock {
    return { request in
        return request.URL?.absoluteString == self.WordPressComOAuthTokenUrl
    }
}
```

Then you can just do:

```swift
stub(isOauthTokenRequest()) { request in
    let stubPath = OHPathForFile("WordPressOAuthClientSuccess.json", self.dynamicType)
    return fixture(stubPath!, headers: ["Content-Type":"application/json"])
}
```